sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -169,6 +170,8 @@ case class Literal protected (value: Any, dataType: DataType)
 
   override def toString: String = if (value != null) value.toString else "null"
 
+  override def hashCode(): Int = 42 * dataType.hashCode() + Objects.hashCode(value)
```

Comment:
Make sense.

I can't see any rules like that http://www.scalastyle.org/rules-0.8.0.html#org_scalastyle_scalariform_EqualsHashCodeChecker and I think that is understandable as hashCode and equals should really be defined together.